docs/architecture.md

```diff
@@ -1,188 +1,260 @@
 
-The architecture diagram from above can be intimidating at first, this
-section explains it step by step.
+On a new cluster, the Kubewarden components defined are:
 
-### Default Policy Server
+<!--TODO:How are CRDs shown in the diagram-->
 
-On a fresh new cluster, the Kubewarden components defined are its Custom
-Resource Definitions, the `kubewarden-controller` Deployment and a `PolicyServer`
-Custom Resource named `default`.
+- the Custom Resource Definitions (CRD)
+- the `kubewarden-controller` Deployment
+- a `PolicyServer` Custom Resource named `default`.
 
 ![Defining the first ClusterAdmissionPolicy resource](/img/architecture_sequence_01.png)
 
-`kubewarden-controller` notices the default `PolicyServer` resource and, as a result of that,
+<!--TODO:For the next paragraph. What type of architecture diagram is this? In the
+diagram, which is `policy-server`. The aquamarine box with 'Kubewarden Policy
+Server'? Is 'Deployment' capitalization correct? What does 'notices' mean
+exactly? For the diagram, what's the difference between solid box, dashed box,
+white box, aquamarine box? What are the arrows showing, should they be
+labelled?-->
+
+When the `kubewarden-controller` notices the default `PolicyServer` resource,
 it creates a Deployment of the `policy-server` component.
 
-As stated above, Kubewarden works as a Kubernetes Admission Webhook. Kubernetes
-dictates that all the Webhook endpoints must be secured with TLS.
-`kubewarden-controller` takes care of setting up this secure communication
-by doing these steps:
+Kubewarden works as a Kubernetes Admission Webhook.
+Kubernetes specifies using TLS to secure all Webhook endpoints.
+The `kubewarden-controller` sets up this secure communication
+by:
+
+1. Generating a self-signed Certificate Authority
+1. Use this CA to generate a TLS certificate key for the `policy-server` Service.
+
+<!--The capitalization on Secret is ok? Maybe in ``?-->
 
-1. Generate a self-signed Certificate Authority
-1. Use this CA to generate a TLS certificate and a TLS key for the
-   `policy-server` Service.
+These objects are all stored as Secret resources in Kubernetes.
 
-All these objects are stored into Kubernetes as Secret resources.
+<!--TODO:Next paragraph. `policy-server` is in the aquamarine box, right? I
+don't think the K8s ClusterIP Service is shown in the diagram anywhere right?
+Should it be?-->
 
-Finally, `kubewarden-controller` will create the `policy-server`
-Deployment and a Kubernetes ClusterIP Service to expose it inside of
-the cluster network.
+Finally, `kubewarden-controller` creates the `policy-server` Deployment
+and a Kubernetes ClusterIP Service
+to expose it inside the cluster network.
 
 ### Defining the first policy
 
-This chart shows what happens when the first policy bounded to the default `policy-server` is defined inside of the
-cluster:
+<!--TODO:First mention of 'bound'. I'd like to define what binding or to
+be bound is, how it happens, where, etc?-->
```

Comment:
I think the word "bound" wants to tell is that the policy must define on which policy server it will run. It's not possible to run the policy in multiples policy servers. If not defined, the `default` policy server will be choose. I'm referring to the policy definition. Not the wasm module.

 In other words, you can have different policies with the same wasm module (even with the same settings) running in multiple servers. But you cannot have a single policy definition that will run in multiple policy servers.